modules/thermalsensor_generic.c

```diff
@@ -0,0 +1,1451 @@
+thermal_sensor_generic_create(const char *name)
+{
+    thermal_sensor_generic_t *self = calloc(1, sizeof *self);
+
+    self->sg_name   = strdup(name);
+    self->sg_temp   = INVALID_TEMPERATURE;
+    self->sg_status = THERMAL_STATUS_INVALID;
+
+    self->sg_temp_cb    = 0;
+    self->sg_temp_path  = 0;
+    self->sg_temp_offs  = 0;
+    self->sg_is_meta    = false;
+
+    self->sg_mode_path    = 0;
+    self->sg_mode_enable  = 0;
+    self->sg_mode_disable = 0;
+
+    self->sg_notify_id    = 0;
+
+    for( int i = 0; i < THERMAL_STATUS_COUNT; ++i ) {
+        self->sg_level[i].sl_mintemp = INVALID_TEMPERATURE;
+        self->sg_level[i].sl_minwait = 0;
+        self->sg_level[i].sl_maxwait = 0;
+    }
+
+    return self;
+}
+
+/** Delete thermal sensor object
+ *
+ * @param self  sensor object, or NULL
+ */
+static void
+thermal_sensor_generic_delete(thermal_sensor_generic_t *self)
+{
+    if( !self )
+        goto EXIT;
+
+    if( self->sg_notify_id ) {
+        g_source_remove(self->sg_notify_id),
+            self->sg_notify_id = 0;
+    }
+
+    free(self->sg_name);
+
+    free(self->sg_temp_path);
+
+    free(self->sg_mode_path);
+    free(self->sg_mode_enable);
+    free(self->sg_mode_disable);
+
+    free(self);
+
+EXIT:
+    return;
+}
+
+/** Check if sensor object is valid
+ *
+ * Check that all necessary values are set and configured paths
+ * exist and are readable/writable as applicable.
+ *
+ * @param self  sensor object, or NULL
+ *
+ * @return true if sensor object is valid, false otherwise
+ */
+static bool
+thermal_sensor_generic_is_valid(thermal_sensor_generic_t *self)
+{
+    bool is_valid = false;
+
+    if( !self )
+        goto EXIT;
+
+    if( !self->sg_name ) {
+        dsme_log(LOG_ERR, PFIX"sensor object without a name");
+        goto EXIT;
+    }
+
+    if( !self->sg_temp_cb ) {
+        dsme_log(LOG_ERR, PFIX"%s: %s",
+                 thermal_sensor_generic_get_name(self),
+                 "no temperature callback defined");
+        goto EXIT;
+    }
+
+    if( !self->sg_temp_path ) {
+        dsme_log(LOG_ERR, PFIX"%s: %s",
+                 thermal_sensor_generic_get_name(self),
+                 "no temperature file path defined");
+        goto EXIT;
+    }
+
+    if( !self->sg_is_meta &&
+        access(self->sg_temp_path, R_OK) != 0 ) {
+        dsme_log(LOG_ERR, PFIX"%s: %s: %m",
+                 thermal_sensor_generic_get_name(self), self->sg_temp_path);
+        goto EXIT;
+    }
+
+    if( self->sg_mode_path ) {
+        if( access(self->sg_mode_path, W_OK) != 0 ) {
+            dsme_log(LOG_ERR, PFIX"%s: %s: %m",
+                     thermal_sensor_generic_get_name(self),
+                     self->sg_mode_path);
+            goto EXIT;
+        }
+
+        if( !self->sg_mode_enable ) {
+            dsme_log(LOG_ERR, PFIX"%s: %s",
+                     thermal_sensor_generic_get_name(self),
+                     "enable value not defined");
+            goto EXIT;
+        }
+
+#if 0
```

Comment:
If we are not going to use that, remove the whole `#if 0`..`#endif` block (both the `then` and `else` blocks).